// @kennethormandy/react-flipcard/lib/components/LabelWrapper.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var LabelWrapper = function LabelWrapper(props) {
  return _react2.default.createElement(
    'label',
    {
      style: {
        display: 'flex',
        flexWrap: 'wrap',
        flex: 'auto',
        alignItems: 'center',
        paddingLeft: '1em',
        paddingRight: '1em',
        marginBottom: '1em'
      } },
    _react2.default.createElement(
      'div',
      { style: { textAlign: 'left', width: '50%' } },
      props.label
    ),
    props.secondaryLabel ? _react2.default.createElement(
      'div',
      {
        style: {
          textAlign: 'right',
          width: '50%',
          fontVariantNumeric: 'tabular-nums'
        } },
      props.secondaryLabel
    ) : null,
    _react2.default.createElement(
      'div',
      { style: { width: '100%', paddingTop: '0.75em' } },
      props.children
    )
  );
};

LabelWrapper.defaultProps = {
  dark: false
};

exports.default = LabelWrapper;
