// @kennethormandy/react-flipcard/lib/components/Slider.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// Styles and approach is based on Rebass
// Available under MIT License https://git.io/v9U64
var Slider = function Slider(props) {
  var max = props.max || 100;
  var min = props.min || 0;
  var percent = (props.value - min) / (max - min) * 100;
  var backgroundImage = props.fill ? 'linear-gradient(90deg, currentcolor, currentcolor ' + percent + '%, transparent ' + percent + '%)' : '';

  var styles = {
    boxSizing: 'border-box',
    display: 'block',
    width: '100%',
    margin: 0,
    // marginTop: scale[1],
    cursor: 'pointer',
    color: 'inherit',
    backgroundImage: backgroundImage,
    backgroundClip: 'content-box',
    height: 6,
    borderRadius: 999,
    WebkitAppearance: 'none',
    appearance: 'none'
  };

  var inputProps = Object.assign({}, props, {
    className: 'Slider-input',
    type: 'range',
    style: styles,
    dark: undefined,
    fill: undefined
  });

  return _react2.default.createElement(
    'div',
    { className: 'Slider ' + (props.dark ? 'Slider--dark' : '') },
    _react2.default.createElement('input', inputProps)
  );
};

Slider.defaultProps = {
  dark: false
};

exports.default = Slider;
